### ISSUE TITLE
[BUG] Scikit-learn 1.4.0 breaks _ProbaScorer

### ISSUE THREAD

**Issue by noamzbr** [deepchecks/deepchecks#2724]
**Describe the bug**
scikit-learn 1.4.0 breaks _ProbaScorer

```
91from sklearn.metrics._scorer import _BaseScorer, _ProbaScorer
92 ImportError: cannot import name '_ProbaScorer' from 'sklearn.metrics._scorer'
```

That's why we limit it in https://github.com/deepchecks/deepchecks/pull/2723

**ToDo**
Have the scorer work for this new version, possibly some kind of switch case

**Comment by j-adamczyk** [comment]
Any news on this? Scikit-learn 1.3 is really quite outdated, and with lots of good new stuff in 1.5 and 1.6 it's really hard for me to justify to keep the old version only to use DeepChecks

**Comment by JKL98ISR** [comment]
Fixed in 0.19.1
